pandas/algos.pyx

```diff
@@ -1348,6 +1348,56 @@ cdef inline float64_t _median_linear(float64_t* a, int n):
     return result
 
 
+cdef int64_t _checked_add_scalars(int64_t a, int64_t b) except -1:
+    """
+    Performs the addition of two int64 integers but checks
+    that they do not result in overflow first.
+
+    Parameters
+    ----------
+    a : first addend.
+    b : second addend.
+
+    Returns
+    -------
+    sum : The sum of a + b.
+
+    Raises
+    ------
+    OverflowError if a + b exceeds the maximum int64 value.
+    """
+    if np.iinfo(np.int64).max - b < a:
```

Comment:
Use the numpy c-api version of this so it's a compile time constant `NPY_MAX_INT64`